### PR TITLE
chore(ci): user friendly release

### LINKED ISSUE #5681: Tracking issue of RFC User-friendly Release
**Motivation**

In the past, only specific binaries were included in the Databend release. Understanding how to operate and maintain Databend often requires extensive discussion.

This proposal aims to improve this process by providing some simple tips and default configs to help make the process painless for users.

**RFC**

[User-friendly Release](https://databend.rs/doc/contributing/rfcs/user-friendly-releases)

**Roadmap**

- [ ] add more comments to the configuration files
- [ ] write a readme.txt
- [ ] update the release workflow

### PR DESCRIPTION
**Pull request by PsiACE**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

- add a readme for release
- only release a databend-*.tar.gz , with bin and configs

## Changelog

- Build/Testing/CI

## Related Issues

Fixes #5681

### CODE REVIEW

File: .github/actions/publish_binary/databend-release-readme.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+- Documentation: https://databend.rs/doc
+- Code Browser: https://github.com/datafuselabs/databend
+
+We'd love to hear from you. If you have any questions about Databend or want to suggest a feature, reach out to us on GitHub: 
+- Issues: https://github.com/datafuselabs/databend/issues
+- Discussions: https://github.com/datafuselabs/databend/discussions
+
+Follow us on social media and stay tuned for more exciting updates on Databend:
+- Twitter：https://twitter.com/Datafuse_Labs
+- Slack：https://datafusecloud.slack.com/
```
Comment (Xuanwo) [inline .github/actions/publish_binary/databend-release-readme.txt:14]: This link requires the user to have already joined slack account.

How about using https://link.databend.rs/join-slack instead?

File: .github/workflows/databend-release.yml [hunk 2]
```diff
@@ -45,8 +45,13 @@ jobs:
         run: |
           brew install gnu-tar
           sudo /usr/sbin/purge
-          gtar -C ./target/${{ steps.target.outputs.target }}/release -czvf databend-${{ steps.get_version.outputs.VERSION }}-${{ steps.target.outputs.target }}.tar.gz databend-query databend-meta
-          gtar -C ./target/${{ steps.target.outputs.target }}/release -czvf databend-tools-${{ steps.get_version.outputs.VERSION }}-${{ steps.target.outputs.target }}.tar.gz databend-metactl
+          mkdir release/${{ steps.target.outputs.target }} release/${{ steps.target.outputs.target }}/bin release/${{ steps.target.outputs.target }}/configs
+          cp ./target/${{ steps.target.outputs.target }}/release/databend-* release/${{ steps.target.outputs.target }}/bin/
+          rm -f release/${{ steps.target.outputs.target }}/bin/*.d
+          cp ./scripts/ci/deploy/config/databend-query-node-1.toml release/${{ steps.target.outputs.target }}/configs/databend-query.toml
```
Comment (Xuanwo) [inline .github/workflows/databend-release.yml:51]: Is this config suitable for the user's example? Do we need to provide a separate example?

File: .github/workflows/databend-release.yml [hunk 3]
```diff
@@ -45,8 +45,13 @@ jobs:
         run: |
           brew install gnu-tar
           sudo /usr/sbin/purge
-          gtar -C ./target/${{ steps.target.outputs.target }}/release -czvf databend-${{ steps.get_version.outputs.VERSION }}-${{ steps.target.outputs.target }}.tar.gz databend-query databend-meta
-          gtar -C ./target/${{ steps.target.outputs.target }}/release -czvf databend-tools-${{ steps.get_version.outputs.VERSION }}-${{ steps.target.outputs.target }}.tar.gz databend-metactl
+          mkdir release/${{ steps.target.outputs.target }} release/${{ steps.target.outputs.target }}/bin release/${{ steps.target.outputs.target }}/configs
```
Comment (everpcpc) [inline .github/workflows/databend-release.yml:48]:
```suggestion
          mkdir -p release/${{ steps.target.outputs.target }}/{bin,configs}
```

File: .github/workflows/databend-release.yml (hunk 2 above)
Comment (PsiACE) [reply .github/workflows/databend-release.yml:51]: These two configuration files basically cover the normal configuration needs and I think we can start by using it as a basic configuration example.

Another benefit may be that we use these configurations in our tests, so subsequent maintenance will be more timely.

As the deployment documentation is further refined, we will eventually provide some more sensible configurations.

File: .github/actions/publish_binary/databend-release-readme.txt (hunk 1 above)
Comment (PsiACE) [reply .github/actions/publish_binary/databend-release-readme.txt]: Done!

**Review by Xuanwo — APPROVED** [review]

**Review by everpcpc — APPROVED** [review]

**Review by soyeric128 — COMMENTED** [review]
The readme file looks good, thank you.

**Pull request merged**